### PR TITLE
executor: fix invalid converting in TiDB that convert bool values 'true' to int numbers '1' in 'set config'

### LINKED ISSUE #17140: executor: fix invalid converting in TiDB that convert bool values 'true' to int numbers '1' in 'set config' statements
## Development Task
If we process `set config tikv key=true`, TiDB will parser the value `true` to an int number and send JSON data `{'key':1}` instead of `{'key':true}` to TiKV, which is not appropriate.

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve?

Issue Number: close #17140 

Problem Summary: If we process `set config tikv key=true`, TiDB will parser the value `true` to an int number and send JSON data `{'key':1}` instead of `{'key':true}` to TiKV, which is not appropriate.

### What is changed and how it works?
Check the value and if it is a bool value, use `true` and `false` instead of int numbers.

Tests 

- Unit test

### Release note 

- Fix invalid converting in TiDB that convert bool values 'true' to int numbers '1' in 'set config'

### CODE REVIEW

File: executor/set_config.go [hunk 1]
```diff
@@ -17,6 +17,7 @@ import (
 	"bytes"
 	"context"
 	"fmt"
+	"github.com/pingcap/parser/mysql"
```
Comment (SunRunAway) [inline executor/set_config.go:20]: Please format this import

Comment (qw4990) [reply executor/set_config.go]: Fixed.

**Pull request merged**